luarules/gadgets/unit_area_timed_damage.lua

```diff
@@ -12,484 +12,229 @@ function gadget:GetInfo()
-    },
-    ['legnap_scav_napalmbombs'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 15,
-        damage = 30,
-        range = 150,
-        resistance = "test",
-    },
-	['legcom_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 10,
-        damage = 75,
-        range = 100,
-        resistance = "test",
-    },
-    --['legcom_scav_napalmmissile'] = { -- These have been replaced with unit cannon
-    --    ceg = "fire-area-150",
-    --    damageCeg = "burnflamexl",
-    --    time = 10,
-    --    damage = 75,
-    --    range = 100,
-    --    resistance = "test",
-    --},
-	['legcomlvl2_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 10,
-        damage = 75,
-        range = 100,
-        resistance = "test",
-    },
-    --['legcomlvl2_scav_napalmmissile'] = { -- These have been replaced with unit cannon
-    --    ceg = "fire-area-150",
-    --    damageCeg = "burnflamexl",
-    --    time = 10,
-    --    damage = 75,
-    --    range = 100,
-    --    resistance = "test",
-    --},
-	['legcomlvl3_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 10,
-        damage = 150,
-        range = 150,
-        resistance = "test",
-    },
-    --['legcomlvl3_scav_napalmmissile'] = { -- These have been replaced with unit cannon
-    --    ceg = "fire-area-150",
-    --    damageCeg = "burnflamexl",
-    --    time = 10,
-    --    damage = 150,
-    --    range = 150,
-    --    resistance = "test",
-    --},
-	['legcomlvl4_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 10,
-        damage = 150,
-        range = 150,
-        resistance = "test",
-    },
-    --['legcomlvl4_scav_napalmmissile'] = { -- These have been replaced with unit cannon
-    --    ceg = "fire-area-150",
-    --    damageCeg = "burnflamexl",
-    --    time = 10,
-    --    damage = 150,
-    --    range = 150,
-    --    resistance = "test",
-    --},
-	['legperdition_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 15,
-        damage = 175,
-        range = 150,
-        resistance = "test",
-    },
-    ['legperdition_scav_napalmmissile'] = {
-        ceg = "fire-area-150",
-        damageCeg = "burnflamexl",
-        time = 15,
-        damage = 175,
-        range = 150,
-        resistance = "test",
-    },
-}
--- convert weaponname -> weaponDefID
-local TimedDamageWeapons = {}
-for name, params in pairs(TimedDamageWeaponsNames) do
-	if WeaponDefNames[name] then
-		TimedDamageWeapons[WeaponDefNames[name].id] = params
+local function StartTimedArea(x, y, z, params)
+	if not params then return end
+
+	local elevation = Spring.GetGroundHeight(x, z)
+	elevation = elevation < 0 and 0 or elevation
+	if y <= elevation + params.atd_radius * 0.5 then
```

Comment:
What is the significance of half the radius? Is this actually the diameter?

Reply:
I copied that from the current/previous script. An area effect that moves very far to reach the ground might seem to appear out of nowhere for players. The distance conditional might prevent an effect from transferring through a shield to appear the ground below, too. I have no particular thoughts on how to change this, though.

Reply:
(also duh: diameter)